server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -103,6 +103,19 @@ public void register(String aggregationName, List<ValuesSourceType> valuesSource
         }, aggregatorSupplier);
     }
 
+    /**
+     * Register an aggregator that applies to any values source type.  This is a convenience method for aggregations that do not care at all
+     * about the types of their inputs.  Aggregations using this version of registration should not make any other registrations, as the
```

Comment:
Out of curiosity, what would happen if an agg registered "any" as well as a specific type?

Reply:
It's ordered, so if the aggregation called `registerAny` first, that would always be used.  Otherwise, it would have both.  Also, plugins are always registered after core, so if an aggregation calls `registerAny` in core, whatever a plugin registers for it will never be reached.
`registerAny` is really only intended to be used for things that are totally values source agnostic, like ValueCount here.  We could add some defense to reject the any if there's something already registered, but it's harder to check in the other direction.  I suppose we could add a flag for it, but that seems way too special case.